sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -259,6 +269,9 @@ class SymmetricHashJoinStateManager(
           return null
         }
 
+        // Make a copy on value row, as below cleanup logic may update the value row silently.
+        currentValue = currentValue.copy(value = currentValue.value.copy())
```

Comment:
so this is the only place to do copy?

Reply:
Yes. That wasn't necessary for format V1 as the original row was stored into state store, and state store (strictly saying, the implementation of HDFS state store provider) makes sure these rows are copied version.

For other places, it can propagate to the callers outside of state manager, and looks like these callers don't need to copy the row. (It's super tricky for me to determine whether the copy is necessary or not, if the code is not in a simple loop or stream.)

Reply:
After seeing the new changes, I think the first version looks better. The caller sides is nested and we still have necessary copies for v1 format. What do you think? @viirya

Reply:
Yep, also prefer the first approach personally. As the issue was in v2 format, the first version is a straightforward way. 
@cloud-fan typo? `... unnecessary copies for v1 format`

Reply:
OK I'll roll back the change. I'll also leave a commit sha so we can do back and forth depending on the decision.

Reply:
Just reverted the latest commit to leave the history and pick the commit selectively according to the decision.